exoscale/zones.go

```diff
@@ -44,6 +60,22 @@ func (z *zones) GetZoneByProviderID(_ context.Context, _ string) (cloudprovider.
 // name. This method is particularly used in the context of external cloud providers where node initialization must
 // be done outside the kubelets.
 func (z *zones) GetZoneByNodeName(ctx context.Context, nodeName types.NodeName) (cloudprovider.Zone, error) {
+	// first look for a statically-configured override
+	if override := z.p.cfg.Instances.getInstanceOverride(nodeName); override != nil {
```

Comment:
Personally I like to flatten code when possible, flow is easier to understand.
```
override := z.p.cfg.Instances.getInstanceOverride(nodeName)
if override != nil && override.External {
	if len(override.Region) != 0 {
		return cloudprovider.Zone{Region: override.Region}, nil
	}

	return cloudprovider.Zone{Region: "external"}, nil
}
```

Reply:
Also it is more common to use `len(n) > 0` (and not `len(n) != 0`), as `len` cannot return negative value. Same result though.

Reply:
But as in this case `override.Region` is a string, you should compare it with empty string: `override.Region != ""`.

Reply:
Corrected, here and all other identical idioms